### PR TITLE
Disable native image PrintAnalysisCallTree by default, change semantics to 'enabledReports=true'

### PR DESCRIPTION
**Pull request by johnaohara**
…cs to 'enabledReports=true'

Reduces heap allocation and increase build speed of a default native image build

### CODE REVIEW

**Comment by johnaohara** [comment]
This PR resolves #2241 but also retains current behaviour, i.e. if users currently define `<disableReports>true</disableReports>` report generation will be disabled

File: devtools/maven/src/main/java/io/quarkus/maven/NativeImageMojo.java [hunk 1]
```diff
@@ -333,7 +337,10 @@ public void accept(ConfigBuilder configBuilder) {
                     configs.add("quarkus.native.debug-symbols", debugSymbols.toString());
                 }
                 if (disableReports != null) {
-                    configs.add("quarkus.native.disable-reports", disableReports.toString());
+                    configs.add("quarkus.native.enable-reports", new Boolean(!disableReports).toString());
```
Comment (gsmet) [inline devtools/maven/src/main/java/io/quarkus/maven/NativeImageMojo.java]: Is it safe? Could `disableReports` be null?

**Review by gsmet — CHANGES_REQUESTED** [review]
Added a small question.

File: devtools/maven/src/main/java/io/quarkus/maven/NativeImageMojo.java (hunk 1 above)
Comment (johnaohara) [reply devtools/maven/src/main/java/io/quarkus/maven/NativeImageMojo.java]: `disableReports` null on line 340? there is a null check on line 339

Comment (gsmet) [reply devtools/maven/src/main/java/io/quarkus/maven/NativeImageMojo.java]: Oops sorry :).

**Review by gsmet — APPROVED** [review]

**Comment by Sanne** [comment]
could we avoid the double negation? I almost misunderstood the option :)

I'd have `<enableReports>` an have it default to `false`.

**Comment by johnaohara** [comment]
@Sanne That is what I have changed it to. Previously the configuration was defined as

**Comment by gsmet** [comment]
Thanks @johnaohara !

**Pull request merged**